src/manifest.c

```diff
@@ -594,8 +617,15 @@ struct manifest *load_mom(int version)
 
 	string_or_die(&filename, "%s/%i/Manifest.MoM", state_dir, version);
 	string_or_die(&url, "%s/%i/Manifest.MoM", content_url, version);
+verify_mom:
 	if (!download_and_verify_signature(url, filename)) {
 		if (sigcheck) {
+			/* cleanup and try one more time, statedir could have got corrupt/stale */
+			if (retried == false) {
+				remove_manifest_files("MoM", version);
+				retried = true;
+				goto verify_mom;
+			}
```

Comment:
I'd rather not put it in the sigcheck == false path, but it makes sense to retry if manifest_from_file fails, and to move manifest_from_file above the sigcheck. If we don't care about sigs and loaded the manifest fine, then don't retry.